pkgs/by-name/ch/chirpstack-concentratord/package.nix

```diff
@@ -0,0 +1,130 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  gitUpdater,
+  protobuf,
+}:
+let
+  libloragw-sx1301 = stdenv.mkDerivation rec {
+    pname = "libloragw-sx1301";
+    version = "5.0.1r2";
+
+    src = fetchFromGitHub {
+      owner = "brocaar";
+      repo = "lora_gateway";
+      rev = "59381129a07858a2a91aeffe21cd6a728219cf23";
+      hash = "sha256-K+GCE3eJq/9CgjIPTCBx3dSbDP0QbTiv0QMAbbLZM7s=";
+    };
+
+    makeFlags = [
+      "-e"
+      "-C"
+      "libloragw"
+    ];
+
+    installPhase = ''
+      mkdir -p $out/{lib,include/libloragw-sx1301}
+      cp libloragw/libloragw.a $out/lib/libloragw-sx1301.a
+      cp libloragw/inc/* $out/include/libloragw-sx1301
+    '';
+  };
+
+  libloragw-sx1302 = stdenv.mkDerivation rec {
+    pname = "libloragw-sx1302";
+    version = "2.1.0r7";
+
+    src = fetchFromGitHub {
+      owner = "brocaar";
+      repo = "sx1302_hal";
+      rev = "c3d99009556fdfe273c3a53306082ef181333c7a";
+      hash = "sha256-Ue3C7XbaCv/d0NIMclPSE+qcQpxw70DVhXlnpFACzhY=";
+    };
+
+    makeFlags = [
+      "-e"
+      "-C"
+      "libloragw"
+    ];
+
+    preBuildPhases = [ "buildDeps" ];
+
+    installPhase = ''
+      mkdir -p $out/{lib,include/libloragw-sx1302}
```

Comment:
If you are overriding `configurePhase`, `buildPhase`, `checkPhase`, `installPhase` or any other phase, you should not forget about explicitly running pre and post-phase hooks like [their](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L953) [original](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1008) [definitions](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1037) [do](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1078).

It is generally expected that an appropriate pre-phase hook (e.g. `preBuild`) hook will run at the beginning of a phase (e.g. `buildPhase`) and post-phase hook (e.g. `postBuild`) will run at the end. Hooks are normally ran as a part of a phase so if you override a phase as a whole, you will need to add `runHook hookName` calls (e.g. `runHook preBuild`) manually.

Having phases run pre/post-phase hooks is important because many setup hooks insert their own code into them – omitting a hook might therefore prevent some setup hooks required for proper functionality of a package from running. Additionally, hooks are often inserted by developers into the package expression and by users when overriding a package using `overrideAttrs`. Not running them can thus cause confusion why their code is not executed.

## Examples

### Before

```nix
  installPhase = ''
    your commands
  '';
```

### After

```nix
  installPhase = ''
    runHook preInstall

    your commands

    runHook postInstall
  '';
```

Source: https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/missing-phase-hooks.md

Reply:
Fixed in ea05670e75d4